feat(api-reference): client v2 modal

amritk opened the pull request:
## Problem

Its not quite done yet but we can start playing around + check the code


To do:
- [x] test all the specs 
- [x] remaining config properties to add to the test:
  - [x] auth
    - [x] preferredSecurityScheme + selectedScopes
  - [x] baseServerUrl
  - [x] proxyUrl
  - [x] theme
  - [x] onRequestSent
  - [x] onBeforeRequest
  - [x] onDocumentSelect
  - [x] onShowMore
  - [x] onSidebarClick
  - [x] documentDownloadType
  - [x] hideDownloadButton
  - [x] hideTestRequestButton
  - [x] expandAllModelSections
  - [x] expandAllResponses

punted to fast follow:
- [ ] persist auth + selected client + server
- [ ] onSpecUpdate - hasn't worked for some time

- the e2e test failures are just sub pixel diffs
- CDN tests are green, bolt now shows complex auth as opposed to simple auth before (upgrade)

## Checklist

- [x] I explained why the change is needed.
- [x] I added a changeset. <!-- pnpm changeset -->
- [x] I added tests.
- [ ] I updated the documentation.

<!--
  Use semantic PR titles:

    fix(api-client): crashes when API returns null
    ^   ^            ^
    |   |            |
    |   |            |____ subject
    |   |_________________ package
    |_____________________ type of change

  Read more: https://github.com/scalar/scalar/blob/main/CONTRIBUTING.md
-->






<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Introduces the Client v2 modal and refactors request/auth flows to support it.
> 
> - **Client v2 modal integration**: New modal factory (`createApiClientModal`) with event bus support, routing helpers, and authentication configuration merge; modal wires into operation view and collection pages.
> - **Auth/security revamp**: Replaces `selectedSecurity` usage with computed `selectedSecuritySchemes`; adds `getSecurityRequirements`, `getSelectedSecurity`, `getSecuritySchemes`, and `mergeAuthConfig` helpers; OAuth2 defaults redirect URI; adds secret coercion (`convertSecuritySchemeSecrets`) and merged schemes typing/exports.
> - **Plugin API change**: `beforeRequest` hook now receives/returns `{ request }`; all callers/tests updated; `sendRequest` and helpers adapted.
> - **Request building improvements**: `buildRequest` now accepts `selectedSecuritySchemes`; `build-request-security` simplified to take resolved schemes; `build-request-body` unwraps proxied File and sets filename; better example/param derivation and form/table handling; small URL/content-type fixes.
> - **Code samples and UI tweaks**: Operation code sample uses event bus, nullable server, stricter props, improved error logging; auth selector label tweaks and optional sections; request/body/table UX copy updates ("Select File").
> - **Routing/operation helpers**: Adds `getSelectedServer` and consolidates operation security/server selection logic; exposes useful helpers/exports across v2 modules.
> - **Tests**: Broad updates/additions across helpers, components, and flows to cover new behavior.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 9fb2b0e888e1c9bbe881ae5281b22c39a2716f29. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

hanspagel commented:
same as before, but variables in the server URL should use:

* a given value, or:
* the default, or:
* empty string

and *not* use the {variable} string

<img width="782" height="356" alt="Screenshot 2026-01-05 at 10 11 31" src="https://github.com/user-attachments/assets/429555a8-ffc3-481d-ae44-ca3fd635abcf" />

hanspagel commented:
the request body for the DELETE request should be "None"

but it’s "other"

and nitpick: it’s "other" in the label and "Other" in the dropdown

<img width="1033" height="909" alt="Screenshot 2026-01-05 at 10 13 41" src="https://github.com/user-attachments/assets/defb53a6-c5ae-4ac6-ba9e-6e7d9c82acb8" />

hanspagel commented:
files are sent as "[object File]" string

<img width="371" height="154" alt="Screenshot 2026-01-05 at 10 19 09" src="https://github.com/user-attachments/assets/6f3f3e44-5663-4f23-8212-23c839fdfcc0" />

hanspagel commented:
I tried the Petstore 2.0 example, and it doesn’t show the fields in the request body:

good:

<img width="579" height="487" alt="Screenshot 2026-01-05 at 10 21 20" src="https://github.com/user-attachments/assets/da6e4401-4deb-4a33-90a6-a86710aea0e7" />

no good:

<img width="577" height="460" alt="Screenshot 2026-01-05 at 10 21 58" src="https://github.com/user-attachments/assets/c80f9142-6f92-4c20-a9d9-c90fbb95994a" />

amritk commented:
Alright I'm going to fix the tests now, @hanspagel I think I addressed all of your issues except:
- server variable handling, I think it matches what we have right now, we can fix it after!
- the petstore auth issue I get on https://petstore.swagger.io/ as well so its something on their end
